qcodes/instrument_drivers/QuantumDesign/DynaCoolPPMS/private/commandhandler.py

```diff
@@ -0,0 +1,124 @@
+from typing import Any, Tuple, List
+import logging
+from collections import namedtuple
+
+log = logging.getLogger(__name__)
+
+try:
+    import win32com.client
+    import pythoncom
+    from pythoncom import VT_BYREF, VT_R8, VT_I4
+except ImportError as e:
+    message = ('To use the DynaCool Driver, please install win32com.'
+               ' Installation can be done with pip install pypiwin32com')
+    log.exception(message)
+    raise ImportError(message)
+
+
+# Variable types
+_variants = {'VT_R8': win32com.client.VARIANT(VT_BYREF | VT_R8, 0.0),
+             'VT_I4': win32com.client.VARIANT(VT_BYREF | VT_I4, 0)}
+CmdArgs = namedtuple('cmd_and_args', 'cmd args')
+
+
+class CommandHandler:
+    """
+    This is the class that gets called by the server.py
+
+    This class is responsible for making the actual calls into the instrument
+    firmware. The idea is that this class get a SCPI-like string from the
+    server, e.g. 'TEMP?' or 'TEMP 300, 10, 1' and then makes the corresponding
+    MultiVu API call (or returns an error message to the server).
+    """
+
+    def __init__(self, inst_type: str='dynacool') -> None:
+        pythoncom.CoInitialize()
+        client_id = f'QD.MULTIVU.{inst_type.upper()}.1'
+        try:
+            self._mvu = win32com.client.Dispatch(client_id)
+        except pythoncom.com_error:
+            error_mssg = ('Could not connect to Multivu Application. Please '
+                          'make sure that the Multi Application is running.')
+            log.exception(error_mssg)
+            raise ValueError(error_mssg)
+
+        # Hard-code what we know about the MultiVu API
+        self._gets = {'TEMP': CmdArgs(cmd=self._mvu.GetTemperature,
+                                      args=[_variants['VT_R8'],
+                                            _variants['VT_I4']])}
+
+        self._sets = {'TEMP': self._mvu.SetTemperature}
+
+    def preparser(self, cmd_str: str) -> Tuple[CmdArgs, bool]:
+        """
+        Parse the raw SCPI-like input string into a list of strings
+
+        Args:
+            cmd_str: A SCPI-like string, e.g. 'TEMP?' or 'TEMP 300, 0.1, 1'
+
+        Returns:
+            A tuple of a CmdArgs tuple and a bool indicating whether this was
+            a query
+        """
+        def err_func():
+            return 'ERROR: unknown command'
+
+        cmd_head = cmd_str[:4]
+
+        if cmd_head not in set(self._gets.keys()).union(set(self._sets.keys())):
+            cmd = err_func
+            args: List[Any] = []
+            is_query = False
+
+        elif cmd_str.endswith('?'):
+            cmd = self._gets[cmd_head].cmd
+            args = self._gets[cmd_head].args
+            is_query = True
+        else:
+            cmd = self._sets[cmd_head]
+            args = list(float(arg) for arg in cmd_str[5:].split(', '))
+            is_query = False
+
+        output = (CmdArgs(cmd=cmd, args=args), is_query)
+
+        return output
```

Comment:
could you write `return CmdArgs(cmd=cmd, args=args), is_query` for readability (i.e. remove `output`)?